training/src/app/chuck-noris/joke-list/joke-list.component.ts

```diff
@@ -17,14 +21,24 @@ import { animate, state, style, transition, trigger } from '@angular/animations'
 	],
 })
 
-export class JokeListComponent implements OnInit {
+export class JokeListComponent implements OnInit{
 	@Input() public jokeList: Joke[];
 	public jokeListTable: MatTableDataSource<Joke> = new MatTableDataSource();
 	public columnsToDisplay: string[]              = ['id', 'category', 'created_at'];
 	public expandedElement: Joke;
+	public untilDestroy: Subject<void>                             = new Subject<void>();
+
+	constructor(private store: Store<AppState>) {
+	}
 
 	ngOnInit(): void {
-		this.jokeListTable.data = this.jokeList;
+		this.store.select(selectJokeList).subscribe((jokeList: Joke[]) => {
+			this.jokeListTable.data = jokeList;
```

Comment:
you are already getting this data from @Input() no?

Reply:
deleted the input part the data didn't pass from there properly